src/openforms/js/components/admin/form_design/logic/actions/Actions.js

```diff
@@ -105,28 +107,62 @@ const ActionFetchFromService = ({action, errors, onChange}) => {
     setIsModalOpen(false);
   };
 
+  const formContext = useContext(FormContext);
+
+  let serviceFetchConfigFromVar =
+    _.cloneDeep(formContext.formVariables.find(element => element.key === action.variable))
+      ?.serviceFetchConfiguration || undefined;
+
+  if (serviceFetchConfigFromVar) {
+    if (!Array.isArray(serviceFetchConfigFromVar.headers)) {
+      serviceFetchConfigFromVar.headers = Object.entries(serviceFetchConfigFromVar.headers || []);
+    }
+
+    if (!Array.isArray(serviceFetchConfigFromVar.queryParams)) {
+      serviceFetchConfigFromVar.queryParams = Object.entries(
+        serviceFetchConfigFromVar.queryParams || []
+      );
+    }
+
+    switch (serviceFetchConfigFromVar.dataMappingType) {
+      case 'JsonLogic':
+        serviceFetchConfigFromVar.jsonLogicExpression = serviceFetchConfigFromVar.mappingExpression;
+        break;
+      case 'jq':
+        serviceFetchConfigFromVar.jqExpression = serviceFetchConfigFromVar.mappingExpression;
+        break;
+      default:
+        serviceFetchConfigFromVar.jqExpression = serviceFetchConfigFromVar.mappingExpression;
+    }
+  }
+
   return (
     <>
       <DSLEditorNode errors={errors.variable}>
-        {/*
-          TODO: should we filter this to only allow user defined variables?
-
-          const filter = variable => variable.source === 'user_defined'
-        */}
-        <VariableSelection name="variable" value={action.variable} onChange={onChange} />
-      </DSLEditorNode>
-      <DSLEditorNode errors={errors.action?.value}>
-        {/* TODO: this element loses state on change of the variable sibling right above*/}
-        {/* TODO: in #2661 we're building a nicer UI/UX to configure the service fetch action */}
-        <input
-          name="action.value"
+        <VariableSelection
+          name="variable"
+          value={action.variable}
           onChange={onChange}
-          value={action.action.value}
-          placeholder="ServiceFetchConfiguration id"
-          type="number"
-          min="1"
+          // Only values of user defined values can be set
+          filter={variable => variable.source === 'user_defined'}
         />
       </DSLEditorNode>
+      <DSLEditorNode errors={errors.action?.value}>
+        <span>
+          <b>
+            <FormattedMessage
+              description="Currently selected service fetch configuration label"
+              defaultMessage="Fetch configuration:"
+            />
+          </b>
+          &nbsp;
+          {serviceFetchConfigFromVar?.name ||
+            intl.formatMessage({
+              description: 'No service fetch configuration configured yet message',
+              defaultMessage: '(not configured yet)',
+            })}
+        </span>
```

Comment:
let's give this a proper class name and do the styling with CSS instead of relying on `<b>` or `<strong>`. Additionally, the text and button are not properly vertically aligned (centered) because the span gets some bottom and right margins that shouldn't be there.